common/compat.h

```diff
@@ -170,6 +170,14 @@ typedef int pid_t;
 #define STDIN_FILENO 0
 #define STDOUT_FILENO 1
 
+/* microsoft Windows uses ISO definitions, not POSIX anymore */
+#define read(fd, buf, len) _read(fd, buf, (unsigned int)(len))
+#define write(fd, buf, len) _write(fd, buf, (unsigned int)(len))
+
+#else
+/* all OS except Windows assumes that closesocket() is close() */
+
+#define closesocket(s) close(s)
```

Comment:
hmm, I would like to keep the code aligned to POSIX, not to the winsock API.

Reply:
The issue is to wrap the calls to close on a filedescriptor vs a socket descriptor. How to support it while keeping the same name ?  The only option I know is to use a name that does not conflict with the other names.

What's about closenet () that would be close () on Unix and closesocket  () on Windows ? 

https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-closesocket

Reply:
My question is rather: would it be really possible to add a compatibility at such level? I even doubt that we can use the same function prototype of `closesocket` for both Unix and Windows, because `closesocket` takes an unsigned integer, while `close` takes a signed integer:
https://docs.microsoft.com/en-us/windows/win32/winsock/socket-data-type-2

If it is not too much burden, I would probably separate the code path completely with `#ifdef`.

Reply:
I believe I could design using,

SOCKET fh = socket( AF_INET, SOCK_STREAM, 0 );
int fd = _open_osfhandle(fh);